proto/minder/v1/minder.proto

```diff
@@ -2559,70 +2559,73 @@ message ListEvaluationHistoryResponse {
+        string details = 2;
 
-message EvaluationHistoryRule {
-    // name is the name of the rule instance.
-    string name = 1;
+        // created_at is the timestamp of creation of this evaluation
+        google.protobuf.Timestamp evaluated_at = 3;
+    }
 
-    // type is the name of the rule type.
-    string type = 2;
+    message Remediation {
+        // status is one of (success, error, failure, skipped, not available)
+        // not using enums to mirror the behaviour of the existing API contracts.
+        string status = 1;
 
-    // profile is the name of the profile which contains the rule.
-    string profile = 3;
-}
+        // details contains optional details about the remediation.
+        // the structure and contents are remediation specific, and are subject to change.
+        string details = 2;
+    }
 
-message EvaluationHistoryStatus {
-    // status is one of (success, error, failure, skipped)
-    // not using enums to mirror the behaviour of the existing API contracts.
-    string status = 1;
+    message Alert {
+        // status is one of (on, off, error, skipped, not available)
+        // not using enums to mirror the behaviour of the existing API contracts.
+        string status = 1;
 
-    // details contains optional details about the evaluation.
-    // the structure and contents are rule type specific, and are subject to change.
-    string details = 2;
-}
+        // details contains optional details about the alert.
+        // the structure and contents are alert specific, and are subject to change.
+        string details = 2;
+    }
 
-message EvaluationHistoryRemediation {
-    // status is one of (success, error, failure, skipped, not available)
-    // not using enums to mirror the behaviour of the existing API contracts.
-    string status = 1;
+    // entity contains details of the entity which was evaluated.
+    EvaluationHistory.Entity entity = 1;
 
-    // details contains optional details about the remediation.
-    // the structure and contents are remediation specific, and are subject to change.
-    string details = 2;
-}
+    // rule contains details of the rule which the entity was evaluated against.
+    EvaluationHistory.Rule rule = 2;
 
-message EvaluationHistoryAlert {
-    // status is one of (on, off, error, skipped, not available)
-    // not using enums to mirror the behaviour of the existing API contracts.
-    string status = 1;
+    // status contains the evaluation status.
+    EvaluationHistory.Status status = 3;
+
+    // alert contains details of the alerts for this evaluation.
+    EvaluationHistory.Alert alert = 4;
 
-    // details contains optional details about the alert.
-    // the structure and contents are alert specific, and are subject to change.
-    string details = 2;
+    // remediation contains details of the remediation for this evaluation.
+    EvaluationHistory.Remediation remediation = 5;
```

Comment:
Do we want a single `status` / `alert` / `remediation` for each entry?  It feels like we'll end up repeating a lot of the envelope (`rule`, `entity`) many times when querying history.  I'd sort of prefer to see something like this:

```proto
message HistoryResult {
  Timestamp start = 1;
  uint32 occurrences = 2;
  string status = 3;  // Include both rule eval and alert / remediate result
  // Temp, while we work out schema
  string status_detail = 4;
  string alert_detail = 5;
  string remediate_detail = 6;
}

message EntityRuleEvaluationHistory {
  EntityInfo entity = 1;
  RuleInfo rule = 2;
  repeated HistoryResult history = 3;
}

message ListEvaluationHistoryResult {
  Pagination pagination = 1;
  repeated HistoryResult = 2;
}
```

Reply:
I am confused by this comment, both `ListEvaluationHistoryResult` and `EntityRuleEvaluationHistory` embed a `repeated HistoryResult`, that's a typo, right?

I don't know how to make progress on this, I'd be happy to discuss if we want to change message structure.

Reply:
Oops, I meant for `ListEvaluationHistoryResult` to embed a repeated `EntityRuleEvaluationHistory`.  Updated.

Reply:
It turns out that we're trying to present something different (a log-history based view of history), so my suggestion doesn't really work.